fastMatcher.go

```diff
@@ -11,23 +11,29 @@ type slotData struct {
 	size  int
 }
 
+var emptySlotData slotData
+
 type FastMatcher struct {
-	def     MatchDef
-	slots   []slotData
-	buckets *binTreeState
-	tokens  jsonTokenizer
+	def      MatchDef
+	slots    []slotData
+	numSlots int
```

Comment:
Why copy numSlots into the matcher?  Why not simply use `matcher.def.numSlots`?

Reply:
True. Done